web/tests/filter.test.js

```diff
@@ -2204,3 +2204,59 @@ run_test("equals", () => {
         ),
     );
 });
+
+run_test("adjusted_terms_if_moved", () => {
+    // should return null for non-stream messages
+    let raw_terms = [{operator: "channel", operand: "Foo"}];
+    let message = {type: "private"};
+    let result = Filter.adjusted_terms_if_moved(raw_terms, message);
+    assert.strictEqual(result, null);
+
+    // should return null if no terms are changed
+    raw_terms = [{operator: "channel", operand: "general"}];
+    message = {type: "stream", display_recipient: "general", topic: "discussion"};
+    result = Filter.adjusted_terms_if_moved(raw_terms, message);
+    assert.strictEqual(result, null);
+
+    // should adjust channel term to match message's display_recipient
+    raw_terms = [{operator: "channel", operand: "random"}];
+    message = {type: "stream", display_recipient: "general", topic: "discussion"};
+    let expected = [{operator: "channel", operand: "general"}];
+    result = Filter.adjusted_terms_if_moved(raw_terms, message);
+    assert.deepStrictEqual(result, expected);
+
+    // should adjust topic term to match message's topic
+    raw_terms = [{operator: "topic", operand: "random"}];
+    message = {type: "stream", display_recipient: "general", topic: "discussion"};
+    expected = [{operator: "topic", operand: "discussion"}];
+    result = Filter.adjusted_terms_if_moved(raw_terms, message);
+    assert.deepStrictEqual(result, expected);
```

Comment:
These first few tests seem like they're verifying a rather unexpected result; do we want this logic to apply to a search that doesn't have both channel and topic operators?

Reply:
Probably worth discussing this question in #frontend for visibility; I don't think it's a blocker to merging this commit, since you're just adding tests for existing behavior in this commit, but maybe you can start a thread?